include/deal.II/lac/block_sparsity_pattern.h

```diff
@@ -968,7 +985,7 @@ BlockDynamicSparsityPattern::column_number(const size_type    row,
 
   size_type c             = 0;
   size_type block_columns = 0; // sum of n_cols for all blocks to the left
-  for (unsigned int b = 0; b < columns; ++b)
+  for (unsigned int b = 0; b < this->n_block_cols(); ++b)
```

Comment:
Should the end condition be in a separate variable so that we don't call `n_block_cols()` repeatedly (and similarly for all other `for`-loops here)?

Reply:
Perhaps, but I am assuming `n_block_cols()` will be inlined - its a function which returns a constant. I can do it if you think its worthwhile but I tried to consistently use the accessor functions everywhere.

Right now `rows()` and `cols()` don't return constants, but they will when this class is upgraded to inherit from `SparsityPatternBase` from the other PR (which is why those methods look so weird right now).